packages/modules/src/BatchedSessionRouterModule.ts

```diff
@@ -0,0 +1,290 @@
+import { ISessionStorage, SessionSearchParam, SessionStatus } from './interfaces/ISessionStorage'
+
+export class BatchedSessionRouterModule extends BaseValidationModule {
+  version: ModuleVersion = 'V1_0_0'
+  moduleAddress!: string
+  sessionManagerModuleAddress!: string
+  nodeClient!: INodeClient
+  merkleTree!: MerkleTree
+  sessionStorageClient!: ISessionStorage
+
+  /**
+   * This constructor is private. Use the static create method to instantiate SessionKeyManagerModule
+   * @param moduleConfig The configuration for the module
+   * @returns An instance of SessionKeyManagerModule
+   */
+  private constructor(moduleConfig: BatchedSessionKeyManagerModuleConfig) {
+    super(moduleConfig)
+  }
+
+  /**
+   * Asynchronously creates and initializes an instance of SessionKeyManagerModule
+   * @param moduleConfig The configuration for the module
+   * @returns A Promise that resolves to an instance of SessionKeyManagerModule
+   */
+  public static async create(
+    moduleConfig: BatchedSessionKeyManagerModuleConfig
+  ): Promise<BatchedSessionRouterModule> {
+    const instance = new BatchedSessionRouterModule(moduleConfig)
+
+    if (moduleConfig.moduleAddress) {
+      instance.moduleAddress = moduleConfig.moduleAddress
+    } else if (moduleConfig.version) {
+      const moduleAddr = BATCHED_SESSION_ROUTER_MODULE_ADDRESSES_BY_VERSION[moduleConfig.version]
+      if (!moduleAddr) {
+        // throw new Error(`Invalid version ${moduleConfig.version}`)
+        Logger.error(`Invalid version ${moduleConfig.version}`)
+        instance.moduleAddress = DEFAULT_BATCHED_SESSION_ROUTER_MODULE
+      }
+      instance.moduleAddress = moduleAddr
+      instance.version = moduleConfig.version as ModuleVersion
+    }
+
+    instance.sessionManagerModuleAddress =
+      moduleConfig.sessionManagerModuleAddress ?? DEFAULT_SESSION_KEY_MANAGER_MODULE
+    instance.nodeClient = new NodeClient({
+      txServiceUrl: moduleConfig.nodeClientUrl ?? NODE_CLIENT_URL
+    })
+
+    if (!moduleConfig.storageType || moduleConfig.storageType === StorageType.LOCAL_STORAGE) {
+      instance.sessionStorageClient = new SessionLocalStorage(moduleConfig.smartAccountAddress)
+    } else {
+      throw new Error('Invalid storage type')
+    }
+
+    const existingSessionData = await instance.sessionStorageClient.getAllSessionData()
+    const existingSessionDataLeafs = existingSessionData.map((sessionData) => {
+      const leafDataHex = hexConcat([
+        hexZeroPad(ethers.utils.hexlify(sessionData.validUntil), 6),
+        hexZeroPad(ethers.utils.hexlify(sessionData.validAfter), 6),
+        hexZeroPad(sessionData.sessionValidationModule, 20),
+        sessionData.sessionKeyData
+      ])
+      return ethers.utils.keccak256(leafDataHex)
+    })
+
+    instance.merkleTree = new MerkleTree(existingSessionDataLeafs, keccak256, {
+      sortPairs: true,
+      hashLeaves: false
+    })
+
+    return instance
```

Comment:
Let SessionKeyManager module handle the merkle tree state. Like the onchain module, this module also can be made to not store any state related to sessions and just route the request to SessionKeyManager module.

In create method, lets take an instance of SessionKeyManager module to enable this.

Coz in current state, if there's any change in merkle tree logic or storage of sessions, we'll need to change the same logic at multiple places. If you are making it easy for developers to use, then in that case we should create an instance of SessionKeyManager module inside this class and use that instance instead of repeating the same logic here.

Reply:
I feel we could remove the `merkleTree` instance all together. We always fetch the local storage and update and create new tree instance.

It is only being used in `signUserOpHash`, so if we are okay to fetch the data again in that method we can remove this.

Reply:
when generating proof we will need this instance. are you suggesting to fetch everything from localstorage create a tree and get proof?

Reply:
create an instance of SessionKeyManager module inside this class and use that instance - for us to create an instance I will also need SA address